app/src/webkit/java/org/mozilla/focus/web/WebViewProvider.java

```diff
@@ -67,6 +70,26 @@ private static void configureSettings(Context context, WebSettings settings) {
         settings.setAllowFileAccess(false);
 
         settings.setBlockNetworkImage(appSettings.shouldBlockImages());
+
+        appSettings.subscribeToBlockImagesPreference(new Settings.BooleanPrefListener() {
+            final WeakReference<WebView> webViewWeakReference = new WeakReference<>(webView);
+
+            @Override
+            public void onPrefChanged(final boolean blockImages) {
+                final WebView webView = webViewWeakReference.get();
+                if (webView != null) {
+                    webView.getSettings().setBlockNetworkImage(blockImages);
+
+                    if (blockImages) {
+                        // Images that are already cached will still be rendered even after enabling
+                        // this setting. To avoid confusing anyone who is actively testing this setting
+                        // it's probably best to clear the cache to make sure absolutely no images
+                        // are shown. (The cache will be cleared when erase()'ing anyway.)
+                        webView.clearCache(true);
```

Comment:
Existing pages aren't reloaded until you explicitly reload, so this isn't too terrible - but yeah, I'm also not convinced either way.